packages/manager/src/queries/events.ts

```diff
@@ -0,0 +1,110 @@
+import { APIError, ResourcePage } from '@linode/api-v4/lib/types';
+import {
+  useQuery,
+  useInfiniteQuery,
+  useQueryClient,
+  InfiniteData,
+} from 'react-query';
+import {
+  Entity,
+  Event,
+  getEvents,
+  markEventSeen,
+} from '@linode/api-v4/lib/account';
+import { isInProgressEvent } from 'src/store/events/event.helpers';
+
+export interface EntityEvent extends Omit<Event, 'entity'> {
+  entity: Entity;
+}
+
+const queryKey = 'events';
+
+export const useEventsPolling = (eventHandler?: (event: Event) => void) => {
+  useQuery<Event[], APIError[]>(
+    [queryKey, 'polling'],
+    async () => {
+      const events = await requestUnseenEvents();
+      await markCompletedEventsAsSeen(events);
+      return events;
+    },
+    {
+      refetchInterval: 5000,
+      retryDelay: 5000,
+      refetchOnMount: false,
+      refetchOnWindowFocus: false,
+      onSuccess: (events) =>
+        eventHandler && events.reverse().forEach(eventHandler),
+    }
+  );
+};
+
+export const useEvents = () => {
```

Comment:
Can we make this hook's name more specific?

```suggestion
export const useEventsInfiniteQuery = () => {
```

Reply:
Should we enforce this naming scheme (i.e., '-Query' after all query hooks and '-Mutation' after all mutation hooks) across the app?